app/file_sender.py

```diff
@@ -158,11 +159,24 @@ def start_file_sender(readiness_queue):
 
 def copy_files_to_sftp(file_paths: Collection[Path], remote_directory):
     with sftp.SftpUtility(remote_directory) as sftp_client:
-        logger.info('Copying files to SFTP remote', sftp_directory=sftp_client.sftp_directory)
+        logger.info('Copying files to SFTP remote and then GCP Bucket',
+                    sftp_directory=sftp_client.sftp_directory, gcp_bucket=Config.SENT_PRINT_FILE_BUCKET)
         for file_path in file_paths:
             sftp_client.put_file(local_path=str(file_path), filename=file_path.name)
-        logger.info(f'All {len(file_paths)} files successfully written to SFTP remote',
-                    sftp_directory=sftp_client.sftp_directory)
+            write_file_to_bucket(file_path)
+
+        logger.info(f'All {len(file_paths)} files successfully written to SFTP remote and gcp bucket',
+                    sftp_directory=sftp_client.sftp_directory, gcp_bucket=Config.SENT_PRINT_FILE_BUCKET)
+
+
+def write_file_to_bucket(file_path):
+    if len(Config.SENT_PRINT_FILE_BUCKET) == 0:
+        logger.warn(f'SENT_PRINT_FILE_BUCKET set to empty')
```

Comment:
Warning message should probably say explicitly that it's skipping the bucket upload?

Reply:
I don't think it's a warning. If we've configured it to not upload to bucket, then that's normal expected behaviour. I would either log debug, or info at most, but really it shouldn't log noise at all if it's just doing what it's told to do and it's not erroring or doing anything unusual.

Reply:
Will reword the warning msg.  Couldn't decide whether it should be a warning or info,  the reason I went for warning is that it's expected behaviour for it to upload the files,  if it's not set it means it's running locally (with no GCP bucket) or the config is wrong.  If the config was wrong we wouldn't upload the files and wouldn't know about it either

Reply:
I guess I'm not too worried about a bunch of warnings getting logged in non-prod environments. Your rationale is sound

Reply:
cheers,  the bucket will/should be created in all gcp environments anyway